### ISSUE TITLE
Is the audit log permission configurable?

### ISSUE THREAD

**Issue by kangzhiqin** [kubernetes/kubernetes#120203]
The default permission of the log file to which audit logs are being written is 600. The permission of the compressed log file is the same as that of the log file to which audit logs are being written. The default permission is 600.

In some scenarios, for security purposes, you are advised not to have the write permission on compressed log files. You are advised to provide permission configuration parameters.

The code is at line 468 of lumberjack.go.

**Comment by pacoxu** [comment]
The code is here.
https://github.com/kubernetes/kubernetes/blob/eb0afbc7d01541275d34befca9da7512511e0971/staging/src/k8s.io/apiserver/pkg/server/options/audit.go#L518-L528

/sig api-machinery

**Comment by kangzhiqin** [comment]
Thank you for your reply, but what I want to ask for is that the file permissions here are not configurable.

```
// compressLogFile compresses the given log file, removing the
// uncompressed log file if successful.
func compressLogFile(src, dst string) (err error) {
	f, err := os.Open(src)
	if err != nil {
		return fmt.Errorf("failed to open log file: %v", err)
	}
	defer f.Close()

	**fi, err := os_Stat(src)**
	if err != nil {
		return fmt.Errorf("failed to stat log file: %v", err)
	}

	if err := chown(dst, fi); err != nil {
		return fmt.Errorf("failed to chown compressed log file: %v", err)
	}

	// If this file already exists, we presume it was created by
	// a previous attempt to compress the log file.
	**gzf, err := os.OpenFile(dst, os.O_CREATE|os.O_TRUNC|os.O_WRONLY, fi.Mode())**
	if err != nil {
		return fmt.Errorf("failed to open compressed log file: %v", err)
	}
	defer gzf.Close()

	gz := gzip.NewWriter(gzf)

	defer func() {
		if err != nil {
			os.Remove(dst)
			err = fmt.Errorf("failed to compress log file: %v", err)
		}
	}()

	if _, err := io.Copy(gz, f); err != nil {
		return err
	}
	if err := gz.Close(); err != nil {
		return err
	}
	if err := gzf.Close(); err != nil {
		return err
	}

	if err := f.Close(); err != nil {
		return err
	}
	if err := os.Remove(src); err != nil {
		return err
	}

	return nil
}
```

**Comment by jiahuif** [comment]
+1 on `umask` approach. 
awaiting @kangzhiqin 's reply.

**Comment by kangzhiqin** [comment]
> umask

Thank you for your reply, unfortunately `umask `can't solve this problem.

**Comment by kangzhiqin** [comment]
> +1 on `umask` approach. awaiting @kangzhiqin 's reply.

**Comment by jiahuif** [comment]
Did you mean you want to have the original log file and the compressed use different permissions?

**Comment by kangzhiqin** [comment]
> Did you mean you want to have the original log file and the compressed use different permissions?

Yes. For security purposes, the permission of the original file is 600 and that of the compressed file is 400. However, this is not possible at present.
The permission of the original file is the same as that of the compressed file, which is hard-coded and cannot be configured.

**Comment by alexzielenski** [comment]
> The code is at line 468 of lumberjack.go

Is this bug for a third party tool called Lumberjack? If so, I think this issue is filed on the wrong repository

https://github.com/natefinch/lumberjack/blob/4cb27fcfbb0f35cb48c542c5ea80b7c1d18933d0/lumberjack.go#L468-L481

**Comment by kangzhiqin** [comment]
Is there any latest solution to this problem in the community?
